share/wake/lib/system/io.wake

```diff
@@ -106,10 +106,16 @@ def statTypeNumberToPathType (inodeType: Integer): PathType = match inodeType
 
 # Returns the system Stat for a given path
 export target stat (path: Path): Result Stat Error =
+    unsafe_stat path.getPathName
+
+# Returns the system Stat for a given path string. Prefer stat instead
+#
+# Should only be used in rare cases where wake invariants are upheld by the caller
```

Comment:
Might be nice to describe the invariants which are at risk, but since we're expecting people to know what they're doing if they use this, it's probably not required.

Reply:
Honestly this probably isn't all that unsafe of a function. Stat doesn't modify the file so its not a huge risk there. The main issue is that you can extract data about a non-path file which isn't ideal but not the end of the world either